gitlab-runner/src/job.rs

```diff
@@ -63,6 +69,14 @@ pub struct Dependency<'a> {
     dependency: &'a JobDependency,
 }
 
+fn is_retriable_error(err: &ClientError) -> bool {
```

Comment:
Should this be directly on `Error`

Reply:
Maaaybe in the future? But right now this logic is pretty specific to artifact downloads, so we might have a slightly different standard for "retriable" elsewhere.